generator/ServiceClientGeneratorLib/Generators/Marshallers/RestXmlRequestMarshaller.tt

```diff
@@ -736,4 +750,40 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
 <#=new string(' ', (level) * 4)#>					}
 <#+
 	}
+#>
+<#+
+	// xsi is a common prefix in attributes, but it cannot be included in the local name itself as the xmlWriter class will throw an exception. 
+	// Some services may model the xmlName, with the xsi prefix included like it is here https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restXmlWithNamespace/main.smithy#L147
```

Comment:
Small nit: In the GitHub UI, you can right click that line and select `Copy permalink`. 

I don't know how often those protocol tests change, but if they do line 147 on the `main` branch may not be the same.

Reply:
cool tip! didn't know about that. just updated